libs/payments/currency/src/lib/currency.manager.ts

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Injectable } from '@nestjs/common';
+import {
+  CURRENCIES_TO_COUNTRIES,
+  SUPPORTED_PAYPAL_CURRENCIES,
+  VALID_COUNTRY_CODES,
+  VALID_CURRENCY_CODES,
+} from './currency.constants';
+import {
+  CurrencyCodeInvalidError,
+  CountryCodeInvalidError,
+  CurrencyCountryMismatchError,
+  AmountExceedsDigitLimitError,
+} from './currency.error';
+
+@Injectable()
+export class CurrencyManager {
+  constructor() {}
+
+  /**
+   * Verify that provided source country and plan currency are compatible with
+   * valid currencies and countries as listed in constants
+   *
+   * @param currency Currency of customer
+   * @param country Country of customer
+   * @returns True if currency is compatible with country, else throws error
+   */
+  isCurrencyCompatibleWithCountry(
+    currency: string | null | undefined,
+    country: string | null | undefined
+  ): boolean {
+    if (!currency) throw new CurrencyCodeInvalidError(currency);
+    if (!country) throw new CountryCodeInvalidError(country);
+
+    const currencyToCountryMap: Record<string, string[]> = {
+      ...CURRENCIES_TO_COUNTRIES,
+    };
+
+    if (
+      !VALID_CURRENCY_CODES.includes(currency) ||
+      !SUPPORTED_PAYPAL_CURRENCIES.includes(currency) ||
+      !currencyToCountryMap.hasOwnProperty(currency)
+    ) {
+      throw new CurrencyCodeInvalidError(currency);
+    }
+
+    if (!VALID_COUNTRY_CODES.includes(country)) {
+      throw new CountryCodeInvalidError(country);
+    }
+
+    if (!currencyToCountryMap[currency].includes(country))
+      throw new CurrencyCountryMismatchError(currency, country);
+
+    return true;
+  }
+
+  /*
+   * Convert amount in cents to paypal AMT string.
+   * We use Stripe to manage everything and plans are recorded in an AmountInCents
+   * But PayPal AMT field requires a string, as documented here: https://developer.paypal.com/docs/nvp-soap-api/do-reference-transaction-nvp/#payment-details-fields
+   */
+  getPayPalAmountStringFromAmountInCents(amountInCents: number): string {
+    // Must be 10 digits or less
+    if (amountInCents.toString().length > 10) {
+      throw new AmountExceedsDigitLimitError(amountInCents);
+    }
+    // Left pad with zeros if necessary, so we always get a minimum of 0.01.
+    const amountAsString = String(amountInCents).padStart(3, '0');
+    const dollars = amountAsString.slice(0, -2);
+    const cents = amountAsString.slice(-2);
+    return `${dollars}.${cents}`;
+  }
```

Comment:
This does seem like something we may want to move to the paypal library rather than having this in the currency manager. Perhaps into the paypal manager?

Reply:
Agreed. It was part of `CurrencyHelper` in payments and isn't actually used at all for this ticket. Since I moved the whole helper over, I thought I might as well update it and add tests. It is pretty PayPal-specific. And It still seems relevant (https://developer.paypal.com/docs/api/payments/v1/#definition-amount). Will update!

Reply:
Moved over to `PayPalManager` along with error to `paypal.error.ts`. Same as above, still have to do some tweaks and finish moving tests over.